transaction/src/model/transaction.rs

```diff
@@ -0,0 +1,145 @@
+use sbor::rust::string::String;
+use sbor::rust::vec::Vec;
+use sbor::*;
+use scrypto::buffer::{scrypto_decode, scrypto_encode};
+use scrypto::crypto::{hash, EcdsaPublicKey, EcdsaSignature, Hash};
+
+use crate::manifest::{compile, CompileError};
+use crate::model::Instruction;
+
+#[derive(Debug, Clone, TypeId, Encode, Decode, PartialEq, Eq)]
+pub enum Network {
+    InternalTestnet,
```

Comment:
Not sure if we need this as a separate field as we could use the V byte in the signature to save a lot of space?